core/src/util/usSharedLibrary.cpp

```diff
@@ -111,6 +110,14 @@ SharedLibrary& SharedLibrary::operator =(const SharedLibrary& other)
   return *this;
 }
 
+bool SharedLibrary::IsSharedLibrary()
+{ // Testing for file extension isn't the most robust way to test
+  // for file type. Furthermore, the existence of this function
+  // is out of place; why test for a shared library if this class
+  // is supposed to represent a shared library?
```

Comment:
We can do a more robust check check later using the object format parser in the 25-module-lifecycle-enhancements branch (I will create a new issue)